test/org/sagebionetworks/bridge/services/ScheduledActivityServiceMockTest.java

```diff
@@ -408,140 +409,155 @@ public void orderActivitieFiltersAndSorts() {
     
     @Test
     public void complexCriteriaBasedScheduleWorksThroughService() throws Exception {
-        String json = TestUtils.createJson("{"+  
-            "'guid':'5fe9029e-beb6-4163-ac35-23d048deeefe',"+
-            "'label':'Voice Activity',"+
-            "'version':4,"+
-            "'modifiedOn':'2016-03-04T20:21:10.487Z',"+
-            "'strategy':{  "+
-                "'type':'CriteriaScheduleStrategy',"+
-                "'scheduleCriteria':[  "+
-                    "{  "+
-                        "'schedule':{"+  
-                            "'scheduleType':'recurring',"+
-                            "'eventId':'enrollment',"+
-                            "'activities':[  "+
-                                "{  "+
-                                    "'label':'Voice Activity',"+
-                                    "'labelDetail':'20 Seconds',"+
-                                    "'guid':'33669208-1d07-4b89-8ec5-1eb5aad6dd75',"+
-                                    "'task':{  "+
-                                        "'identifier':'3-APHPhonation-C614A231-A7B7-4173-BDC8-098309354292',"+
-                                        "'type':'TaskReference'"+
-                                    "},"+
-                                    "'activityType':'task',"+
-                                    "'type':'Activity'"+
+        // As long as time zone is consistent, the right number of tasks will be generated on 
+        // the day of the request, regardless of the hour of the day.
+        executeComplexTestInTimeZone(3, DateTimeZone.forOffsetHours(-7));
+        executeComplexTestInTimeZone(3, DateTimeZone.forOffsetHours(-7));
+        executeComplexTestInTimeZone(23, DateTimeZone.forOffsetHours(8));
+        executeComplexTestInTimeZone(23, DateTimeZone.forOffsetHours(8));
```

Comment:
I think you got these backwards again, I think you meant hour 3 offset -7, hour 23 offset -7, hour 3 offset +8, hour 23 offset +8. Instead, you did hour 3 offset -7 (which converts to 10:00Z) and hour 23 offset +8 (which converts to 15:00Z), and you did each test twice.